Back/dev

echocarriet opened the pull request:
新增項目：
- 調整登入頁顯示密碼icon位置
- 帳戶設定 API 串接、新增重設密按鈕與 Modal。
- 投票項目：顯示 member 所有投票項目、建立新投票 API 串接、編輯投票 API 串接、頁碼 API 串接。

echocarriet committed:
新增時間轉換方法、調整登入頁顯示密碼icon位置

echocarriet committed:
串接帳戶設定API

echocarriet committed:
新增API串接-帳戶設定、新增投票Modal、編輯投票Modal

huan5678 committed:
GIT 提交信息：

功能：實現書籍管理和黑暗模式功能

- 在設置中實現黑暗模式轉換
- 向服務器添加書籍模型、控制器和路由
- 為書籍模型創建數據庫遷移
- 添加新的書籍UI組件
- 實現書籍搜索功能
- 設置JWT身份驗證
- 實現客戶端路由與保護路徑
- 創建用戶設置頁面，包含主題切換
- 修復圖片標籤中缺少的alt屬性
- 修改按鈕顏色以改善對比度
- 更新頁腳佈局並添加社交媒體連結

Signed-off-by: 黃士桓 <huan5678@gmail.com>

huan5678 committed:
[wip]

huan5678 committed:
GIT 提交訊息：

重構：改進驗證的 Vue 彈窗重構

- 擴展 `vee-validate` 導入範圍，以包括 `Field`、`useField`、`useFieldArray`，並在 EditModal Vue 組件中更新 `useForm` 的解構賦值。
- 在 `pollValidationSchema` 中為投票選項添加最小驗證值 1。
- 修改初始投票數據設置，以處理在 EditModal 中最初未提供數據的情況。
- 使用 `vee-validate` 在 EditModal 中實現個別字段定義和錯誤處理。
- 更改 `editPollDataModal` 設置，使用 `props.pollData` 的深拷貝來保持 Vue 組件的響應性。
- 重構 `uploadCoverFile` 和 `changeTag` 函數，以配合 EditModal 中更新的字段結構。
- 調整 `CreateOption` 函數，使用 `useFieldArray` 方法並簡化邏輯。
- 在 ShareModal Vue 組件中，用 `openModal` 響應式引用替換硬編碼的 `openModal` 屬性。
- 更新共享模態視窗的 URL 生成，使用 `window.location.href` 並新增安全上下文檢查，以及剪貼簿 API 的錯誤處理。
- 移除不必要的 `baseUrl` 引用，並在 ShareModal 中使用更適應的 `Url` 引用。
- 對各種模態元素中的 UI 文本進行調整，以反映數據綁定的變更。

Signed-off-by: 黃士桓 <huan5678@gmail.com>

huan5678 committed:
在翻譯 git commit 訊息時，保持原格式的同時還要確保高水準和符合程序員共識的翻譯是很重要的。以下是翻譯：

GIT COMMIT 訊息：

feat: 改善投票管理及 UI 一致性

- 新增投票的刪除功能處理。
- 在 EditModal 組件中重構 props 物件建置，以使用展開運算子。
- 在 utilities 中實現上傳至 Imgur 前的圖片縮放處理。
- 移除 VoteView 組件中不必要的 ref 鉤子和 `inject` 導入。
- 簡化 modal 組件的調用，並傳遞新的 `delPoll` 方法以在 VoteView 組件中刪除投票。
- 替換 VoteView 組件中顯示訊息的 swal 函式庫用法，改用 `useMessageStore`。
- 修正 VoteView 組件中檢索成員投票的分頁查詢參數建置。
- 在 VoteView 組件的各種方法中新增防禦性檢查，並使用 `useMessageStore` 集中訊息處理。
- 執行一般代碼清理，移除未使用的 refs 和導入，處理投票的編輯和刪除功能。
- 引入改進檔案上傳處理的變更，在 DelModal 和 EditModal 組件中都進行了改進。

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息：

feat: 使用 Vue 的 API 重構模態視窗和表單

- 為刪除模態視窗組件添加一個新的 `contentType` 屬性，以便於提示中更好地提供上下文
- 重新構建模態視窗與表單字段的交互，利用 Vue 的 `useField` API 進行表單管理
- 變更刪除提示，使用動態內容類型代替刪除模態視窗中的靜態文本
- 用 Vue 的 `useForm` API 替換手動表單管理，去除重複並簡化表單處理
- 添加新功能來處理表單字段並更新編輯模態視窗以使用結果對象
- 引入新的計算屬性和觀察者來追踪編輯模態視窗內的變化和狀態
- 更新評論視圖以處理用戶互動、刪除操作和 API 請求評論
- 為評論視圖實現分頁並清理模板以簡化用戶介面
- 重構投票視圖以清理數據結構並改善投票編輯的表單提交
- 改善投票視圖表格標頭的可讀性並修正小的文本不一致問題

請記得翻譯所有給定的 git 提交訊息。

Signed-off-by: 黃士桓 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息：

feat：重構分頁與 Vue 組件

- 在 `PaginationView.vue` 中重新排列類屬性，以潔淨的代碼結構
- 為分頁元素添加註釋以增進清晰度
- 修復分頁功能與用戶界面問題
- 修改 `currentPage === totalPage` 檢查，將 `totalPage` 解析為數字
- 為分頁組件中的 `totalPage` 與 `currentPage` 添加預設屬性值
- 新增一個 Vue 組件 `CommentModal.vue` 並調整其相關功能
- 在 `VoteView.vue` 重構 `pagination`、`comments` 與模態組件
- 移除 `AccountView.vue` 中的未使用樣式並改善元素佈局
- 簡化 `CommentView.vue` 中的條件渲染並糾正模態窗口
- 調整類別順序，移除未使用的 delModal，並清理代碼以提高可讀性

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
以下是繁體中文的翻譯：

GIT 提交訊息：

重構：使用組合式 API 改造投票表單處理流程

- 用 `watchEffect` 替換 `onMounted` 進行投票數據表單的初始化。
- 添加一個新的響應式 `pollData` 參考(ref)，以存儲表單數據。
- 更新表單驗證和錯誤處理，使用 `watch` 並設定 `pollData` 物件的深度選項。
- 移除 VeeValidate 中的 `useFieldArray` 和 `useForm`，並將表單及字段處理重构為 Vue 的組合式 API。
- 在新的 `pollData` 參考(ref)結構中實現圖像上傳和選項添加功能。
- 調整 `handleStartPoll` 方法，以在 `pollData` 參考(ref)中設定投票狀態和開始日期。
- 將表單中的 `Field` 組件替換為標準輸入元素。
- 修改選項框的類別，從 `bg-gray-100` 改為 `bg-gray-100 rounded`。
- 更新模板，直接將表單輸入綁定至 `pollData` 參考(ref)屬性。
- 透過 `formErrors` 參考(ref)提供表單驗證反饋，避免錯誤訊息重複。
- 添加邏輯，重置並顯示錯誤訊息以進行表單驗證。
- 使用 `isLoading` 狀態條件渲染加載中的旋轉圖標。
- 添加工具函式來創建、更新和刪除投票，並對新的表單處理程序進行重構。
- 調整 `createNewPoll`、`openEditModal` 和 `updateEditPoll` 函式以支持更新後的表單數據結構，包括標籤映射和 API 標頭。
- 調整 `getMemberPolls`、`createNewPoll` 和 `updateEditPoll` 函式，在操作後獲取和更新投票列表。
- 統一後端各組件視圖中的容器類別使用，以實現風格一致性。

Signed-off-by: 黃士桓 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息：

功能：重構 Vue 組件中的投票處理

- 新增一個用於確認模態框的 Vue 組件 `ConformModal.vue`
- 移除 `EditPollModal.vue` 及其關聯代碼
- 將 `EditModal.vue` 重命名為 `PollModal.vue` 並更新相關引用
- 在 `PollModal.vue` 中引入預設圖片常量並改進圖片錯誤處理
- 在 `PollModal.vue` 中實現啟動和結束投票的功能
- 在 `VoteView.vue` 中將多個 `show*` ref 變量替換為單一的 `showModal` ref 變量
- 調整 `VoteView.vue` 中開啟各種模態框的事件處理，以使用新的 `showModal` ref
- 用 closeModal 函數替換 `VoteView.vue` 中明確的開啟和關閉模態框函數
- 將新的 `ConformModal` 整合進 `VoteView.vue` 中用於投票動作

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
[fix] conflicts

huan5678 committed:
GIT 提交信息：

功能：增強 UI 條件渲染邏輯

- 如果總頁數不為零，在 `PaginationView.vue` 中條件性地渲染分頁。
- 在 `SignupView.vue` 中，將 `useCookie` 導入替換為來自命名空間工具模塊的解構式 `setCookie`。
- 在 `AccountView.vue` 中將新字段 'name' 和 'gender' 引入 `memberPerson` 數據對象並列印含會員詳情的控制台日誌。
- 在 `AccountView.vue` 中防範 `birthday` 為空並正確地從 `memberPerson` 提取它。
- 在 `CommentView.vue` 中增加對 `totalPage` 值的控制台日誌記錄。
- 在 `CommentView.vue` 中增加條件渲染以顯示消息，根據用戶是否有評論顯示不同消息。
- 在 `VoteView.vue` 中增加條件渲染以在沒有開放投票時顯示消息。
- 在沒有投票時使用的模板中移除多餘的 `tbody`，並用 div 和消息替換，在 `VoteView.vue` 中實施。

記得翻譯所給的所有 git 提交信息。

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息:

重構：改善模態界面及投票互動

- 在 `ConformModal.vue` 修正一個打字錯誤，由 `confirmContent` 更正為 `conformContent`。
- 在 `ConformModal.vue` 中更改模態視窗關閉按鈕的滑鼠懸停背景顏色，從紅色變為更淺的灰色。
- 更新確認模態視窗的訊息，使用 `<div v-html>` 指令來插入 HTML 內容，在 `ConformModal.vue` 中進行調整。
- 為 `PollModal.vue` 新增一個預設的封面圖片 URL。
- 在 `PollModal.vue` 中實現封面圖片上傳的錯誤處理，並在出錯時設定預設的封面圖片。
- 重構 `handlePollAction` 方法，從元件中移除並更新其呼叫位置，直接調用傳遞作為屬性的 `propsHandlePollAction`，以正確的投票數據參數在 `PollModal.vue` 中呼叫該方法。
- 重構選項影像上傳的錯誤處理，以適應新的參數結構，在 `PollModal.vue` 中進行更新。
- 新增狀態變量 `pollActionType` 和 `pollActionTarget`，並對 `VoteView.vue` 中開始和結束投票的 `handlePollAction` 函數進行重構。
- 在 `VoteView.vue` 中將 `confirmContent` 參照替換為 `conformContent`，以符合其在 `ConformModal` 元件中的正確使用。

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息：

重構：重構 VoteView 模態視窗及投票邏輯

- 在 VoteView 的 handlePoll 函式中增加一個 closeModal 呼叫
- 傳遞 pollActionTarget 給 handlePoll，而不是在 VoteView 內的 ConformModal 組件中直接使用它

Signed-off-by: huan5678 <huan5678@gmail.com>

eee317 commented:
測試後有發現以下的問題，再麻煩留意一下，感謝

上傳選項的圖片，上傳之後沒有顯示在上面
<img width="596" alt="image" src="https://github.com/sp1daTomm/SelectWave/assets/38075054/20038826-f4f5-45d2-89ae-f37235ca3a13">

設定開始日期之後，建立投票的按鈕無法點選，導致無法建立投票
不確定觸發點是什麼，有時候好像是主視覺照片沒有傳，或是選項照片傳了沒有回應就可能會觸發，導致按鈕一直無法點選，只能關閉視窗重來
<img width="594" alt="image" src="https://github.com/sp1daTomm/SelectWave/assets/38075054/223fa50f-6d2c-487a-b327-fd2f5ba22e9b">

huan5678 committed:
GIT 提交訊息：

feat: 增強用戶介面一致性與投票處理

- `App.vue` 中更新大頭貼樣式，使用圓角邊框和灰色邊框顏色。
- 在 `PollModal.vue` 的文件上傳後，增加對 `pollData.value.options` 的日誌記錄。
- 通過條件分割投票開始日期，修正潛在的未定義錯誤。
- 移除硬編碼的狀態檢查，現在直接使用原始投票狀態值。
- 根據投票狀態，為投票封面圖片添加透明度變化。
- 更改文字顏色修改，以便考慮所有輸入欄位的投票狀態。
- 修正 `uploadOptionFile` 函式呼叫中的事件參考，以正確處理文件上傳。
- 調整邊框和環狀樣式，使其與投票選項元素中的禁用狀態樣式一致。
- 在 `AccountView.vue` 成功更新會員資訊後刷新頁面。

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息：

重構：重構模態組件並改善日期處理

- 移除 AddPollModal.vue 組件及其關聯的模態
- 重構 ConformModal.vue 組件以使用 defineEmits 並移除不必要的 props
- 重構 DelModal.vue 組件以使用 defineEmits 並移除不必要的 props
- 引入 formatDate 函數並在 PollModal.vue 中清理日期處理
- 在 PollModal.vue 中改用 emit 方式來關閉模態和提交資料
- 更新 ShareModal.vue 以使用 defineEmits 和 emit 來進行 closeModal 操作
- 調整 VoteView.vue 以使用新的 emit 方法處理投票操作和提交，並移除老方法
- 對多個模態組件進行一般代碼清理和重構

記得將所有給定的 git 提交訊息翻譯。

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息：

feat: 用新內容和佈局重構「關於」頁面

- 在 `about.json` 中更新團隊成員部分，加入新成員的詳情、角色、圖片、簡介，以及社交媒體連結。
- 把 LinkedIn 和 Twitter 的社交媒體引用改為 GitHub、Instagram 和電子郵件，並換上相應的圖標。
- 在 `about.json` 中將公司創始人換成新的個體及其頭銜。
- 在 `AboutView.vue` 中，向 useForm 解構中添加 `meta`。
- 修改 `AboutView.vue` 中的頭部和版塊佈局以提升響應性。
- 調整團隊部分的成員圖片展現和社交媒體圖標。
- 改進新聞訂閱表單的樣式，並增加表單驗證反饋。
- 使用新的圖片和佈局更新服務部分，以達到更好的視覺呈現效果。

Signed-off-by: 黃士桓 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息：

重構：改善路由和 UI 組件代碼庫

- 更新「Voting」路由以包含「:id」參數用於動態路由
- 重構 AccountView 的 HTML，將多行代碼整合為較少的行數，同時不改變功能
- 在 AccountView 中，用 router-link 替換按鈕並將文字從「重設密碼」更改為「更改密碼」
- 在 ChangePasswordView 中，移除不必要的導入，用導入的方法替換 cookie 工具的使用，並用自訂訊息處理替換「swal」庫的使用
- 修改 CommentView 和 VoteView 的路由，以便使用帶有動態 ID 參數的「Voting」名稱
- 在 VoteView 中添加一個新函數「linkToPollDetail」以處理重定向到投票詳情頁面
- 對 Vue 組件文件進行各種代碼風格和格式化改善，而不改變功能

Signed-off-by: 黃士桓 <huan5678@gmail.com>
